{"matches":[{"vulnerability":{"id":"GHSA-h95x-26f3-88hr","dataSource":"https://github.com/advisories/GHSA-h95x-26f3-88hr","namespace":"github:language:python","severity":"High","urls":[],"description":"js2py allows remote code execution","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-28397","epss":0.54777,"percentile":0.97889,"date":"2025-06-02"}],"fix":{"versions":[],"state":"not-fixed"},"advisories":[],"risk":44.643255},"relatedVulnerabilities":[{"id":"CVE-2024-28397","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-28397","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/Marven11","https://github.com/Marven11/CVE-2024-28397-js2py-Sandbox-Escape","https://github.com/Marven11","https://github.com/Marven11/CVE-2024-28397-js2py-Sandbox-Escape"],"description":"An issue in the component js2py.disable_pyimport() of js2py up to v0.74 allows attackers to execute arbitrary code via a crafted API call.","cvss":[{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:L/I:L/A:L","metrics":{"baseScore":5.3,"exploitabilityScore":1.9,"impactScore":3.4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-28397","epss":0.54777,"percentile":0.97889,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"js2py","version":"0.74"}},"found":{"versionConstraint":"<=0.74 (python)","vulnerabilityID":"GHSA-h95x-26f3-88hr"}}],"artifact":{"id":"b305fe1a47c44c47","name":"js2py","version":"0.74","type":"python","locations":[{"path":"/lib/python3.12/site-packages/Js2Py-0.74.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/Js2Py-0.74.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/Js2Py-0.74.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/Js2Py-0.74.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.12/site-packages/Js2Py-0.74.dist-info/top_level.txt","accessPath":"/lib/python3.12/site-packages/Js2Py-0.74.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:piotr_dabkowski_project:python-js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piotr_dabkowski_project:python_js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piotr_dabkowskiproject:python-js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piotr_dabkowskiproject:python_js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piotr_dabkowski_project:js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piodrus_project:python-js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piodrus_project:python_js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piotr_dabkowski:python-js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piotr_dabkowski:python_js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piotr_dabkowskiproject:js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piodrusproject:python-js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piodrusproject:python_js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:python-js2py:python-js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:python-js2py:python_js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:python_js2py:python-js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:python_js2py:python_js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piodrus_project:js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piotr_dabkowski:js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piodrus:python-js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piodrus:python_js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piodrusproject:js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:python:python-js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:python:python_js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:js2py:python-js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:js2py:python_js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:python-js2py:js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:python_js2py:js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:piodrus:js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:python:js2py:0.74:*:*:*:*:*:*:*","cpe:2.3:a:js2py:js2py:0.74:*:*:*:*:*:*:*"],"purl":"pkg:pypi/js2py@0.74","upstreams":[]}},{"vulnerability":{"id":"GHSA-45x7-px36-x8w8","dataSource":"https://github.com/advisories/GHSA-45x7-px36-x8w8","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Prefix Truncation Attack against ChaCha20-Poly1305 and Encrypt-then-MAC aka Terrapin","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.9,"exploitabilityScore":2.3,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-48795","epss":0.67021,"percentile":0.98447,"date":"2025-06-02"}],"fix":{"versions":["3.4.0"],"state":"fixed"},"advisories":[],"risk":36.526445},"relatedVulnerabilities":[{"id":"CVE-2023-48795","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-48795","namespace":"nvd:cpe","severity":"Medium","urls":["http://packetstormsecurity.com/files/176280/Terrapin-SSH-Connection-Weakening.html","http://seclists.org/fulldisclosure/2024/Mar/21","http://www.openwall.com/lists/oss-security/2023/12/18/3","http://www.openwall.com/lists/oss-security/2023/12/19/5","http://www.openwall.com/lists/oss-security/2023/12/20/3","http://www.openwall.com/lists/oss-security/2024/03/06/3","http://www.openwall.com/lists/oss-security/2024/04/17/8","https://access.redhat.com/security/cve/cve-2023-48795","https://arstechnica.com/security/2023/12/hackers-can-break-ssh-channel-integrity-using-novel-data-corruption-attack/","https://bugs.gentoo.org/920280","https://bugzilla.redhat.com/show_bug.cgi?id=2254210","https://bugzilla.suse.com/show_bug.cgi?id=1217950","https://crates.io/crates/thrussh/versions","https://filezilla-project.org/versions.php","https://forum.netgate.com/topic/184941/terrapin-ssh-attack","https://git.libssh.org/projects/libssh.git/commit/?h=stable-0.10&id=10e09e273f69e149389b3e0e5d44b8c221c2e7f6","https://github.com/NixOS/nixpkgs/pull/275249","https://github.com/PowerShell/Win32-OpenSSH/issues/2189","https://github.com/PowerShell/Win32-OpenSSH/releases/tag/v9.5.0.0p1-Beta","https://github.com/TeraTermProject/teraterm/commit/7279fbd6ef4d0c8bdd6a90af4ada2899d786eec0","https://github.com/TeraTermProject/teraterm/releases/tag/v5.1","https://github.com/advisories/GHSA-45x7-px36-x8w8","https://github.com/apache/mina-sshd/issues/445","https://github.com/connectbot/sshlib/commit/5c8b534f6e97db7ac0e0e579331213aa25c173ab","https://github.com/connectbot/sshlib/compare/2.2.21...2.2.22","https://github.com/cyd01/KiTTY/issues/520","https://github.com/drakkan/sftpgo/releases/tag/v2.5.6","https://github.com/erlang/otp/blob/d1b43dc0f1361d2ad67601169e90a7fc50bb0369/lib/ssh/doc/src/notes.xml#L39-L42","https://github.com/erlang/otp/releases/tag/OTP-26.2.1","https://github.com/golang/crypto/commit/9d2ee975ef9fe627bf0a6f01c1f69e8ef1d4f05d","https://github.com/hierynomus/sshj/issues/916","https://github.com/janmojzis/tinyssh/issues/81","https://github.com/jtesta/ssh-audit/commit/8e972c5e94b460379fe0c7d20209c16df81538a5","https://github.com/libssh2/libssh2/pull/1291","https://github.com/mkj/dropbear/blob/17657c36cce6df7716d5ff151ec09a665382d5dd/CHANGES#L25","https://github.com/mscdex/ssh2/commit/97b223f8891b96d6fc054df5ab1d5a1a545da2a3","https://github.com/mwiede/jsch/compare/jsch-0.2.14...jsch-0.2.15","https://github.com/mwiede/jsch/issues/457","https://github.com/mwiede/jsch/pull/461","https://github.com/net-ssh/net-ssh/blob/2e65064a52d73396bfc3806c9196fc8108f33cd8/CHANGES.txt#L14-L16","https://github.com/openssh/openssh-portable/commits/master","https://github.com/paramiko/paramiko/issues/2337","https://github.com/proftpd/proftpd/blob/0a7ea9b0ba9fcdf368374a226370d08f10397d99/RELEASE_NOTES","https://github.com/proftpd/proftpd/blob/d21e7a2e47e9b38f709bec58e3fa711f759ad0e1/RELEASE_NOTES","https://github.com/proftpd/proftpd/blob/master/RELEASE_NOTES","https://github.com/proftpd/proftpd/issues/456","https://github.com/rapier1/hpn-ssh/releases","https://github.com/ronf/asyncssh/blob/develop/docs/changes.rst","https://github.com/ronf/asyncssh/tags","https://github.com/ssh-mitm/ssh-mitm/issues/165","https://github.com/warp-tech/russh/releases/tag/v0.40.2","https://gitlab.com/libssh/libssh-mirror/-/tags","https://groups.google.com/g/golang-announce/c/-n5WqVC18LQ","https://groups.google.com/g/golang-announce/c/qA3XtxvMUyg","https://help.panic.com/releasenotes/transmit5/","https://jadaptive.com/important-java-ssh-security-update-new-ssh-vulnerability-discovered-cve-2023-48795/","https://lists.debian.org/debian-lts-announce/2023/12/msg00017.html","https://lists.debian.org/debian-lts-announce/2024/01/msg00013.html","https://lists.debian.org/debian-lts-announce/2024/01/msg00014.html","https://lists.debian.org/debian-lts-announce/2024/04/msg00016.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/33XHJUB6ROFUOH2OQNENFROTVH6MHSHA/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/3CAYYW35MUTNO65RVAELICTNZZFMT2XS/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/3JIMLVBDWOP4FUPXPTB4PGHHIOMGFLQE/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/3YQLUQWLIHDB5QCXQEX7HXHAWMOKPP5O/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/6Y74KVCPEPT4MVU3LHDWCNNOXOE5ZLUR/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/APYIXIQOVDCRWLHTGB4VYMAUIAQLKYJ3/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/BL5KTLOSLH2KHRN4HCXJPK3JUVLDGEL6/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/C3AFMZ6MH2UHHOPIWT5YLSFV3D2VB3AC/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/CHHITS4PUOZAKFIUBQAQZC7JWXMOYE4B/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/F7EYCFQCTSGJXWO3ZZ44MGKFC5HA7G3Y/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/HYEDEXIKFKTUJIN43RG4B7T5ZS6MHUSP/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/I724O3LSRCPO4WNVIXTZCT4VVRMXMMSG/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/KEOTKBUPZXHE3F352JBYNTSNRXYLWD6P/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/KMZCVGUGJZZVDPCVDA7TEB22VUCNEXDD/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/L5Y6MNNVAPIJSXJERQ6PKZVCIUXSNJK7/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/LZQVUHWVWRH73YBXUQJOD6CKHDQBU3DM/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/MKQRBF3DWMWPH36LBCOBUTSIZRTPEZXB/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/QI3EHAHABFQK7OABNCSF5GMYP6TONTI7/","https://matt.ucc.asn.au/dropbear/CHANGES","https://nest.pijul.com/pijul/thrussh/changes/D6H7OWTTMHHX6BTB3B6MNBOBX2L66CBL4LGSEUSAI2MCRCJDQFRQC","https://news.ycombinator.com/item?id=38684904","https://news.ycombinator.com/item?id=38685286","https://news.ycombinator.com/item?id=38732005","https://nova.app/releases/#v11.8","https://oryx-embedded.com/download/#changelog","https://psirt.global.sonicwall.com/vuln-detail/SNWLID-2024-0002","https://roumenpetrov.info/secsh/#news20231220","https://security-tracker.debian.org/tracker/CVE-2023-48795","https://security-tracker.debian.org/tracker/source-package/libssh2","https://security-tracker.debian.org/tracker/source-package/proftpd-dfsg","https://security-tracker.debian.org/tracker/source-package/trilead-ssh2","https://security.gentoo.org/glsa/202312-16","https://security.gentoo.org/glsa/202312-17","https://security.netapp.com/advisory/ntap-20240105-0004/","https://support.apple.com/kb/HT214084","https://thorntech.com/cve-2023-48795-and-sftp-gateway/","https://twitter.com/TrueSkrillor/status/1736774389725565005","https://ubuntu.com/security/CVE-2023-48795","https://winscp.net/eng/docs/history#6.2.2","https://www.bitvise.com/ssh-client-version-history#933","https://www.bitvise.com/ssh-server-version-history","https://www.chiark.greenend.org.uk/~sgtatham/putty/changes.html","https://www.crushftp.com/crush10wiki/Wiki.jsp?page=Update","https://www.debian.org/security/2023/dsa-5586","https://www.debian.org/security/2023/dsa-5588","https://www.freebsd.org/security/advisories/FreeBSD-SA-23:19.openssh.asc","https://www.lancom-systems.de/service-support/allgemeine-sicherheitshinweise#c243508","https://www.netsarang.com/en/xshell-update-history/","https://www.openssh.com/openbsd.html","https://www.openssh.com/txt/release-9.6","https://www.openwall.com/lists/oss-security/2023/12/18/2","https://www.openwall.com/lists/oss-security/2023/12/20/3","https://www.paramiko.org/changelog.html","https://www.reddit.com/r/sysadmin/comments/18idv52/cve202348795_why_is_this_cve_still_undisclosed/","https://www.suse.com/c/suse-addresses-the-ssh-v2-protocol-terrapin-attack-aka-cve-2023-48795/","https://www.terrapin-attack.com","https://www.theregister.com/2023/12/20/terrapin_attack_ssh","https://www.vandyke.com/products/securecrt/history.txt","http://packetstormsecurity.com/files/176280/Terrapin-SSH-Connection-Weakening.html","http://seclists.org/fulldisclosure/2024/Mar/21","http://www.openwall.com/lists/oss-security/2023/12/18/3","http://www.openwall.com/lists/oss-security/2023/12/19/5","http://www.openwall.com/lists/oss-security/2023/12/20/3","http://www.openwall.com/lists/oss-security/2024/03/06/3","http://www.openwall.com/lists/oss-security/2024/04/17/8","https://access.redhat.com/security/cve/cve-2023-48795","https://arstechnica.com/security/2023/12/hackers-can-break-ssh-channel-integrity-using-novel-data-corruption-attack/","https://bugs.gentoo.org/920280","https://bugzilla.redhat.com/show_bug.cgi?id=2254210","https://bugzilla.suse.com/show_bug.cgi?id=1217950","https://crates.io/crates/thrussh/versions","https://filezilla-project.org/versions.php","https://forum.netgate.com/topic/184941/terrapin-ssh-attack","https://git.libssh.org/projects/libssh.git/commit/?h=stable-0.10&id=10e09e273f69e149389b3e0e5d44b8c221c2e7f6","https://github.com/NixOS/nixpkgs/pull/275249","https://github.com/PowerShell/Win32-OpenSSH/issues/2189","https://github.com/PowerShell/Win32-OpenSSH/releases/tag/v9.5.0.0p1-Beta","https://github.com/TeraTermProject/teraterm/commit/7279fbd6ef4d0c8bdd6a90af4ada2899d786eec0","https://github.com/TeraTermProject/teraterm/releases/tag/v5.1","https://github.com/advisories/GHSA-45x7-px36-x8w8","https://github.com/apache/mina-sshd/issues/445","https://github.com/connectbot/sshlib/commit/5c8b534f6e97db7ac0e0e579331213aa25c173ab","https://github.com/connectbot/sshlib/compare/2.2.21...2.2.22","https://github.com/cyd01/KiTTY/issues/520","https://github.com/drakkan/sftpgo/releases/tag/v2.5.6","https://github.com/erlang/otp/blob/d1b43dc0f1361d2ad67601169e90a7fc50bb0369/lib/ssh/doc/src/notes.xml#L39-L42","https://github.com/erlang/otp/releases/tag/OTP-26.2.1","https://github.com/golang/crypto/commit/9d2ee975ef9fe627bf0a6f01c1f69e8ef1d4f05d","https://github.com/hierynomus/sshj/issues/916","https://github.com/janmojzis/tinyssh/issues/81","https://github.com/jtesta/ssh-audit/commit/8e972c5e94b460379fe0c7d20209c16df81538a5","https://github.com/libssh2/libssh2/pull/1291","https://github.com/mkj/dropbear/blob/17657c36cce6df7716d5ff151ec09a665382d5dd/CHANGES#L25","https://github.com/mscdex/ssh2/commit/97b223f8891b96d6fc054df5ab1d5a1a545da2a3","https://github.com/mwiede/jsch/compare/jsch-0.2.14...jsch-0.2.15","https://github.com/mwiede/jsch/issues/457","https://github.com/mwiede/jsch/pull/461","https://github.com/net-ssh/net-ssh/blob/2e65064a52d73396bfc3806c9196fc8108f33cd8/CHANGES.txt#L14-L16","https://github.com/openssh/openssh-portable/commits/master","https://github.com/paramiko/paramiko/issues/2337","https://github.com/proftpd/proftpd/blob/0a7ea9b0ba9fcdf368374a226370d08f10397d99/RELEASE_NOTES","https://github.com/proftpd/proftpd/blob/d21e7a2e47e9b38f709bec58e3fa711f759ad0e1/RELEASE_NOTES","https://github.com/proftpd/proftpd/blob/master/RELEASE_NOTES","https://github.com/proftpd/proftpd/issues/456","https://github.com/rapier1/hpn-ssh/releases","https://github.com/ronf/asyncssh/blob/develop/docs/changes.rst","https://github.com/ronf/asyncssh/tags","https://github.com/ssh-mitm/ssh-mitm/issues/165","https://github.com/warp-tech/russh/releases/tag/v0.40.2","https://gitlab.com/libssh/libssh-mirror/-/tags","https://groups.google.com/g/golang-announce/c/-n5WqVC18LQ","https://groups.google.com/g/golang-announce/c/qA3XtxvMUyg","https://help.panic.com/releasenotes/transmit5/","https://jadaptive.com/important-java-ssh-security-update-new-ssh-vulnerability-discovered-cve-2023-48795/","https://lists.debian.org/debian-lts-announce/2023/12/msg00017.html","https://lists.debian.org/debian-lts-announce/2024/01/msg00013.html","https://lists.debian.org/debian-lts-announce/2024/01/msg00014.html","https://lists.debian.org/debian-lts-announce/2024/04/msg00016.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/33XHJUB6ROFUOH2OQNENFROTVH6MHSHA/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/3CAYYW35MUTNO65RVAELICTNZZFMT2XS/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/3JIMLVBDWOP4FUPXPTB4PGHHIOMGFLQE/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/3YQLUQWLIHDB5QCXQEX7HXHAWMOKPP5O/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/6Y74KVCPEPT4MVU3LHDWCNNOXOE5ZLUR/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/APYIXIQOVDCRWLHTGB4VYMAUIAQLKYJ3/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/BL5KTLOSLH2KHRN4HCXJPK3JUVLDGEL6/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/C3AFMZ6MH2UHHOPIWT5YLSFV3D2VB3AC/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/CHHITS4PUOZAKFIUBQAQZC7JWXMOYE4B/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/F7EYCFQCTSGJXWO3ZZ44MGKFC5HA7G3Y/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/HYEDEXIKFKTUJIN43RG4B7T5ZS6MHUSP/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/I724O3LSRCPO4WNVIXTZCT4VVRMXMMSG/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/KEOTKBUPZXHE3F352JBYNTSNRXYLWD6P/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/KMZCVGUGJZZVDPCVDA7TEB22VUCNEXDD/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/L5Y6MNNVAPIJSXJERQ6PKZVCIUXSNJK7/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/LZQVUHWVWRH73YBXUQJOD6CKHDQBU3DM/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/MKQRBF3DWMWPH36LBCOBUTSIZRTPEZXB/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/QI3EHAHABFQK7OABNCSF5GMYP6TONTI7/","https://matt.ucc.asn.au/dropbear/CHANGES","https://nest.pijul.com/pijul/thrussh/changes/D6H7OWTTMHHX6BTB3B6MNBOBX2L66CBL4LGSEUSAI2MCRCJDQFRQC","https://news.ycombinator.com/item?id=38684904","https://news.ycombinator.com/item?id=38685286","https://news.ycombinator.com/item?id=38732005","https://nova.app/releases/#v11.8","https://oryx-embedded.com/download/#changelog","https://psirt.global.sonicwall.com/vuln-detail/SNWLID-2024-0002","https://roumenpetrov.info/secsh/#news20231220","https://security-tracker.debian.org/tracker/CVE-2023-48795","https://security-tracker.debian.org/tracker/source-package/libssh2","https://security-tracker.debian.org/tracker/source-package/proftpd-dfsg","https://security-tracker.debian.org/tracker/source-package/trilead-ssh2","https://security.gentoo.org/glsa/202312-16","https://security.gentoo.org/glsa/202312-17","https://security.netapp.com/advisory/ntap-20240105-0004/","https://support.apple.com/kb/HT214084","https://thorntech.com/cve-2023-48795-and-sftp-gateway/","https://twitter.com/TrueSkrillor/status/1736774389725565005","https://ubuntu.com/security/CVE-2023-48795","https://winscp.net/eng/docs/history#6.2.2","https://www.bitvise.com/ssh-client-version-history#933","https://www.bitvise.com/ssh-server-version-history","https://www.chiark.greenend.org.uk/~sgtatham/putty/changes.html","https://www.crushftp.com/crush10wiki/Wiki.jsp?page=Update","https://www.debian.org/security/2023/dsa-5586","https://www.debian.org/security/2023/dsa-5588","https://www.freebsd.org/security/advisories/FreeBSD-SA-23:19.openssh.asc","https://www.lancom-systems.de/service-support/allgemeine-sicherheitshinweise#c243508","https://www.netsarang.com/en/xshell-update-history/","https://www.openssh.com/openbsd.html","https://www.openssh.com/txt/release-9.6","https://www.openwall.com/lists/oss-security/2023/12/18/2","https://www.openwall.com/lists/oss-security/2023/12/20/3","https://www.paramiko.org/changelog.html","https://www.reddit.com/r/sysadmin/comments/18idv52/cve202348795_why_is_this_cve_still_undisclosed/","https://www.suse.com/c/suse-addresses-the-ssh-v2-protocol-terrapin-attack-aka-cve-2023-48795/","https://www.terrapin-attack.com","https://www.theregister.com/2023/12/20/terrapin_attack_ssh","https://www.vandyke.com/products/securecrt/history.txt","https://www.vicarius.io/vsociety/posts/cve-2023-48795-detect-openssh-vulnerabilit","https://www.vicarius.io/vsociety/posts/cve-2023-48795-mitigate-openssh-vulnerability"],"description":"The SSH transport protocol with certain OpenSSH extensions, found in OpenSSH before 9.6 and other products, allows remote attackers to bypass integrity checks such that some packets are omitted (from the extension negotiation message), and a client and server may consequently end up with a connection for which some security features have been downgraded or disabled, aka a Terrapin attack. This occurs because the SSH Binary Packet Protocol (BPP), implemented by these extensions, mishandles the handshake phase and mishandles use of sequence numbers. For example, there is an effective attack against SSH's use of ChaCha20-Poly1305 (and CBC with Encrypt-then-MAC). The bypass occurs in chacha20-poly1305@openssh.com and (if CBC is used) the -etm@openssh.com MAC algorithms. This also affects Maverick Synergy Java SSH API before 3.1.0-SNAPSHOT, Dropbear through 2022.83, Ssh before 5.1.1 in Erlang/OTP, PuTTY before 0.80, AsyncSSH before 2.14.2, golang.org/x/crypto before 0.17.0, libssh before 0.10.6, libssh2 through 1.11.0, Thorn Tech SFTP Gateway before 3.4.6, Tera Term before 5.1, Paramiko before 3.4.0, jsch before 0.2.15, SFTPGo before 2.5.6, Netgate pfSense Plus through 23.09.1, Netgate pfSense CE through 2.7.2, HPN-SSH through 18.2.0, ProFTPD before 1.3.8b (and before 1.3.9rc2), ORYX CycloneSSH before 2.3.4, NetSarang XShell 7 before Build 0144, CrushFTP before 10.6.0, ConnectBot SSH library before 2.2.22, Apache MINA sshd through 2.11.0, sshj through 0.37.0, TinySSH through 20230101, trilead-ssh2 6401, LANCOM LCOS and LANconfig, FileZilla before 3.66.4, Nova before 11.8, PKIX-SSH before 14.4, SecureCRT before 9.4.3, Transmit5 before 5.10.4, Win32-OpenSSH before 9.5.0.0p1-Beta, WinSCP before 6.2.2, Bitvise SSH Server before 9.32, Bitvise SSH Client before 9.33, KiTTY through 0.76.1.13, the net-ssh gem 7.2.0 for Ruby, the mscdex ssh2 module before 1.15.0 for Node.js, the thrussh library before 0.35.1 for Rust, and the Russh crate before 0.40.2 for Rust.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.9,"exploitabilityScore":2.3,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-48795","epss":0.67021,"percentile":0.98447,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"paramiko","version":"3.3.1"}},"found":{"versionConstraint":">=2.5.0,<3.4.0 (python)","vulnerabilityID":"GHSA-45x7-px36-x8w8"},"fix":{"suggestedVersion":"3.4.0"}}],"artifact":{"id":"ea5c33ca13c73d0b","name":"paramiko","version":"3.3.1","type":"python","locations":[{"path":"/lib/python3.12/site-packages/paramiko-3.3.1.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/paramiko-3.3.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/paramiko-3.3.1.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/paramiko-3.3.1.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.12/site-packages/paramiko-3.3.1.dist-info/top_level.txt","accessPath":"/lib/python3.12/site-packages/paramiko-3.3.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["LGPL"],"cpes":["cpe:2.3:a:jeff_forcier_project:python-paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff_forcier_project:python_paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff_forcierproject:python-paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff_forcierproject:python_paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python-paramiko:python-paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python-paramiko:python_paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python_paramiko:python-paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python_paramiko:python_paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff_forcier_project:paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff_forcier:python-paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff_forcier:python_paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff_forcierproject:paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff_project:python-paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff_project:python_paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeffproject:python-paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeffproject:python_paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:paramiko:python-paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:paramiko:python_paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python-paramiko:paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python_paramiko:paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff_forcier:paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff_project:paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff:python-paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff:python_paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeffproject:paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:paramiko:paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python:paramiko:3.3.1:*:*:*:*:*:*:*","cpe:2.3:a:jeff:paramiko:3.3.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/paramiko@3.3.1","upstreams":[]}},{"vulnerability":{"id":"CVE-2025-0938","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-0938","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/python/cpython/commit/526617ed68cde460236c973e5d0a8bad4de896ba","https://github.com/python/cpython/commit/90e526ae67b172ed7c6c56e7edad36263b0f9403","https://github.com/python/cpython/commit/a7084f6075c9595ba60119ce8c62f1496f50c568","https://github.com/python/cpython/commit/b8b4b713c5f8ec0958c7ef8d29d6711889bc94ab","https://github.com/python/cpython/commit/d89a5f6a6e65511a5f6e0618c4c30a7aa5aba56a","https://github.com/python/cpython/commit/ff4e5c25666f63544071a6b075ae8b25c98b7a32","https://github.com/python/cpython/issues/105704","https://github.com/python/cpython/pull/129418","https://mail.python.org/archives/list/security-announce@python.org/thread/K4EUG6EKV6JYFIC24BASYOZS4M5XOQIB/","https://security.netapp.com/advisory/ntap-20250314-0002/"],"description":"The Python standard library functions `urllib.parse.urlsplit` and `urlparse` accepted domain names that included square brackets which isn't valid according to RFC 3986. Square brackets are only meant to be used as delimiters for specifying IPv6 and IPvFuture hosts in URLs. This could result in differential parsing across the Python URL parser and other specification-compliant URL parsers.","cvss":[{"source":"cna@python.org","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:H/AT:P/PR:N/UI:N/VC:N/VI:L/VA:N/SC:N/SI:L/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":6.3},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-0938","epss":0.01039,"percentile":0.76414,"date":"2025-06-02"}],"fix":{"versions":["3.9.22","3.10.17","3.11.12","3.12.9","3.13.2","3.14.0a5"],"state":"fixed"},"advisories":[],"risk":0.5870349999999999},"relatedVulnerabilities":[],"matchDetails":[{"type":"cpe-match","matcher":"stock-matcher","searchedBy":{"namespace":"nvd:cpe","cpes":["cpe:2.3:a:python:python:3.13.0:*:*:*:*:*:*:*"],"package":{"name":"python","version":"3.13.0"}},"found":{"vulnerabilityID":"CVE-2025-0938","versionConstraint":"< 3.9.22,>= 3.10.0, < 3.10.17,>= 3.11.0, < 3.11.12,>= 3.12.0, < 3.12.9,>= 3.13.0, < 3.13.2,>= 3.14.0a1, < 3.14.0a5 (unknown)","cpes":["cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"]},"fix":{"suggestedVersion":"3.13.2"}}],"artifact":{"id":"47bfec9b02a55f84","name":"python","version":"3.13.0","type":"binary","locations":[{"path":"/usr/bin/python3.13","accessPath":"/usr/bin/python3.13","annotations":{"evidence":"primary"}}],"language":"","licenses":[],"cpes":["cpe:2.3:a:python_software_foundation:python:3.13.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python:3.13.0:*:*:*:*:*:*:*"],"purl":"pkg:generic/python@3.13.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-5rjg-fvgr-3xxf","dataSource":"https://github.com/advisories/GHSA-5rjg-fvgr-3xxf","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools has a path traversal vulnerability in PackageIndex.download that leads to Arbitrary File Write","cvss":[],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.6206,"date":"2025-06-02"}],"fix":{"versions":["78.1.1"],"state":"fixed"},"advisories":[],"risk":0.32625},"relatedVulnerabilities":[{"id":"CVE-2025-47273","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-47273","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/blob/6ead555c5fb29bc57fe6105b1bffc163f56fd558/setuptools/package_index.py#L810C1-L825C88","https://github.com/pypa/setuptools/commit/250a6d17978f9f6ac3ac887091f2d32886fbbb0b","https://github.com/pypa/setuptools/issues/4946","https://github.com/pypa/setuptools/security/advisories/GHSA-5rjg-fvgr-3xxf","https://lists.debian.org/debian-lts-announce/2025/05/msg00035.html","https://github.com/pypa/setuptools/issues/4946"],"description":"setuptools is a package that allows users to download, build, install, upgrade, and uninstall Python packages. A path traversal vulnerability in `PackageIndex` is present in setuptools prior to version 78.1.1. An attacker would be allowed to write files to arbitrary locations on the filesystem with the permissions of the process running the Python code, which could escalate to remote code execution depending on the context. Version 78.1.1 fixes the issue.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:H/VA:N/SC:N/SI:N/SA:N/E:P/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":7.7},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.6206,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"75.4.0"}},"found":{"versionConstraint":"<78.1.1 (python)","vulnerabilityID":"GHSA-5rjg-fvgr-3xxf"},"fix":{"suggestedVersion":"78.1.1"}}],"artifact":{"id":"33ad8fe548e18c07","name":"setuptools","version":"75.4.0","type":"python","locations":[{"path":"/lib/python3.12/site-packages/setuptools-75.4.0.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/setuptools-75.4.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/setuptools-75.4.0.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/setuptools-75.4.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.12/site-packages/setuptools-75.4.0.dist-info/top_level.txt","accessPath":"/lib/python3.12/site-packages/setuptools-75.4.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:python:setuptools:75.4.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@75.4.0","upstreams":[]}},{"vulnerability":{"id":"CVE-2024-12254","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-12254","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/python/cpython/commit/71e8429ac8e2adc10084ab5ec29a62f4b6671a82","https://github.com/python/cpython/commit/9aa0deb2eef2655a1029ba228527b152353135b5","https://github.com/python/cpython/commit/e991ac8f2037d78140e417cc9a9486223eb3e786","https://github.com/python/cpython/issues/127655","https://github.com/python/cpython/pull/127656","https://mail.python.org/archives/list/security-announce@python.org/thread/H4O3UBAOAQQXGT4RE3E4XQYR5XLROORB/","http://www.openwall.com/lists/oss-security/2024/12/06/1","https://security.netapp.com/advisory/ntap-20250404-0010/"],"description":"Starting in Python 3.12.0, the asyncio._SelectorSocketTransport.writelines()\n method would not \"pause\" writing and signal to the Protocol to drain \nthe buffer to the wire once the write buffer reached the \"high-water \nmark\". Because of this, Protocols would not periodically drain the write\n buffer potentially leading to memory exhaustion.\n\n\n\n\n\nThis\n vulnerability likely impacts a small number of users, you must be using\n Python 3.12.0 or later, on macOS or Linux, using the asyncio module \nwith protocols, and using .writelines() method which had new \nzero-copy-on-write behavior in Python 3.12.0 and later. If not all of \nthese factors are true then your usage of Python is unaffected.","cvss":[{"source":"cna@python.org","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:N/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":8.7},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-12254","epss":0.00144,"percentile":0.35961,"date":"2025-06-02"}],"fix":{"versions":["3.12.9","3.13.2","3.14.0a3"],"state":"fixed"},"advisories":[],"risk":0.11232000000000002},"relatedVulnerabilities":[],"matchDetails":[{"type":"cpe-match","matcher":"stock-matcher","searchedBy":{"namespace":"nvd:cpe","cpes":["cpe:2.3:a:python:python:3.13.0:*:*:*:*:*:*:*"],"package":{"name":"python","version":"3.13.0"}},"found":{"vulnerabilityID":"CVE-2024-12254","versionConstraint":">= 3.12.0, < 3.12.9,>= 3.13.0, < 3.13.2,>= 3.14.0a1, < 3.14.0a3 (unknown)","cpes":["cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"]},"fix":{"suggestedVersion":"3.13.2"}}],"artifact":{"id":"47bfec9b02a55f84","name":"python","version":"3.13.0","type":"binary","locations":[{"path":"/usr/bin/python3.13","accessPath":"/usr/bin/python3.13","annotations":{"evidence":"primary"}}],"language":"","licenses":[],"cpes":["cpe:2.3:a:python_software_foundation:python:3.13.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python:3.13.0:*:*:*:*:*:*:*"],"purl":"pkg:generic/python@3.13.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-8495-4g3g-x7pr","dataSource":"https://github.com/advisories/GHSA-8495-4g3g-x7pr","namespace":"github:language:python","severity":"Medium","urls":[],"description":"aiohttp allows request smuggling due to incorrect parsing of chunk extensions","cvss":[],"epss":[{"cve":"CVE-2024-52304","epss":0.00153,"percentile":0.37217,"date":"2025-06-02"}],"fix":{"versions":["3.10.11"],"state":"fixed"},"advisories":[],"risk":0.0765},"relatedVulnerabilities":[{"id":"CVE-2024-52304","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-52304","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/aio-libs/aiohttp/commit/259edc369075de63e6f3a4eaade058c62af0df71","https://github.com/aio-libs/aiohttp/security/advisories/GHSA-8495-4g3g-x7pr"],"description":"aiohttp is an asynchronous HTTP client/server framework for asyncio and Python. Prior to version 3.10.11, the Python parser parses newlines in chunk extensions incorrectly which can lead to request smuggling vulnerabilities under certain conditions. If a pure Python version of aiohttp is installed (i.e. without the usual C extensions) or `AIOHTTP_NO_EXTENSIONS` is enabled, then an attacker may be able to execute a request smuggling attack to bypass certain firewalls or proxy protections. Version 3.10.11 fixes the issue.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:P/PR:N/UI:N/VC:N/VI:L/VA:N/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":6.3},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-52304","epss":0.00153,"percentile":0.37217,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"aiohttp","version":"3.10.8"}},"found":{"versionConstraint":"<=3.10.10 (python)","vulnerabilityID":"GHSA-8495-4g3g-x7pr"},"fix":{"suggestedVersion":"3.10.11"}}],"artifact":{"id":"23a9da734e9a4097","name":"aiohttp","version":"3.10.8","type":"python","locations":[{"path":"/lib/python3.12/site-packages/aiohttp-3.10.8.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/aiohttp-3.10.8.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/aiohttp-3.10.8.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/aiohttp-3.10.8.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.12/site-packages/aiohttp-3.10.8.dist-info/top_level.txt","accessPath":"/lib/python3.12/site-packages/aiohttp-3.10.8.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["Apache 2"],"cpes":["cpe:2.3:a:python-aiohttp:python-aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python-aiohttp:python_aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python_aiohttp:python-aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python_aiohttp:python_aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:aiohttp:python-aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:aiohttp:python_aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python-aiohttp:aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python_aiohttp:aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python:python-aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python:python_aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:aiohttp:aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python:aiohttp:3.10.8:*:*:*:*:*:*:*"],"purl":"pkg:pypi/aiohttp@3.10.8","upstreams":[]}},{"vulnerability":{"id":"GHSA-q2x7-8rv6-6q7h","dataSource":"https://github.com/advisories/GHSA-q2x7-8rv6-6q7h","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja has a sandbox breakout through indirect reference to format method","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56326","epss":0.00103,"percentile":0.29451,"date":"2025-06-02"}],"fix":{"versions":["3.1.5"],"state":"fixed"},"advisories":[],"risk":0.06592},"relatedVulnerabilities":[{"id":"CVE-2024-56326","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-56326","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/48b0687e05a5466a91cd5812d604fa37ad0943b4","https://github.com/pallets/jinja/releases/tag/3.1.5","https://github.com/pallets/jinja/security/advisories/GHSA-q2x7-8rv6-6q7h"],"description":"Jinja is an extensible templating engine. Prior to 3.1.5, An oversight in how the Jinja sandboxed environment detects calls to str.format allows an attacker that controls the content of a template to execute arbitrary Python code. To exploit the vulnerability, an attacker needs to control the content of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates. Jinja's sandbox does catch calls to str.format and ensures they don't escape the sandbox. However, it's possible to store a reference to a malicious string's format method, then pass that to a filter that calls it. No such filters are built-in to Jinja, but could be present through custom filters in an application. After the fix, such indirect calls are also handled by the sandbox. This vulnerability is fixed in 3.1.5.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:P/PR:L/UI:P/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":5.4},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56326","epss":0.00103,"percentile":0.29451,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.1.4"}},"found":{"versionConstraint":"<=3.1.4 (python)","vulnerabilityID":"GHSA-q2x7-8rv6-6q7h"},"fix":{"suggestedVersion":"3.1.5"}}],"artifact":{"id":"3119b2b2848d353f","name":"jinja2","version":"3.1.4","type":"python","locations":[{"path":"/lib/python3.12/site-packages/jinja2-3.1.4.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/jinja2-3.1.4.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/jinja2-3.1.4.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/jinja2-3.1.4.dist-info/RECORD","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:python-jinja2:python-jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.1.4:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.1.4","upstreams":[]}},{"vulnerability":{"id":"GHSA-gmj9-h825-chq2","dataSource":"https://github.com/advisories/GHSA-gmj9-h825-chq2","namespace":"github:language:python","severity":"High","urls":[],"description":"try/except* clauses could allow bypass RestrictedPython via type confusion bug in the CPython interpreter","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:C/C:H/I:H/A:L","metrics":{"baseScore":7.9,"exploitabilityScore":1.4,"impactScore":6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-22153","epss":0.00083,"percentile":0.25431,"date":"2025-06-02"}],"fix":{"versions":["8.0"],"state":"fixed"},"advisories":[],"risk":0.06391},"relatedVulnerabilities":[{"id":"CVE-2025-22153","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-22153","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/zopefoundation/RestrictedPython/commit/48a92c5bb617a647cffd0dadd4d5cfe626bcdb2f","https://github.com/zopefoundation/RestrictedPython/security/advisories/GHSA-gmj9-h825-chq2"],"description":"RestrictedPython is a tool that helps to define a subset of the Python language which allows to provide a program input into a trusted environment. Via a type confusion bug in versions of the CPython interpreter starting in 3.11 and prior to 3.13.2 when using `try/except*`, RestrictedPython starting in version 6.0 and prior to version 8.0 could be bypassed. The issue is patched in version 8.0 of RestrictedPython by removing support for `try/except*` clauses. No known workarounds are available.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:C/C:H/I:H/A:L","metrics":{"baseScore":7.9,"exploitabilityScore":1.4,"impactScore":6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-22153","epss":0.00083,"percentile":0.25431,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"restrictedpython","version":"7.3"}},"found":{"versionConstraint":">=6.0,<8.0 (python)","vulnerabilityID":"GHSA-gmj9-h825-chq2"},"fix":{"suggestedVersion":"8.0"}}],"artifact":{"id":"7707cf6109c15601","name":"restrictedpython","version":"7.3","type":"python","locations":[{"path":"/lib/python3.12/site-packages/RestrictedPython-7.3.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/RestrictedPython-7.3.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/RestrictedPython-7.3.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/RestrictedPython-7.3.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.12/site-packages/RestrictedPython-7.3.dist-info/top_level.txt","accessPath":"/lib/python3.12/site-packages/RestrictedPython-7.3.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["ZPL 2.1"],"cpes":["cpe:2.3:a:zope_foundation_and_contributors_project:python-restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_foundation_and_contributors_project:python_restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_foundation_and_contributorsproject:python-restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_foundation_and_contributorsproject:python_restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_foundation_and_contributors_project:restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_foundation_and_contributors:python-restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_foundation_and_contributors:python_restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_foundation_and_contributorsproject:restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_foundation_and_contributors:restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:python-restrictedpython:python-restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:python-restrictedpython:python_restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:python_restrictedpython:python-restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:python_restrictedpython:python_restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:python-restrictedpython:restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:python_restrictedpython:restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:restrictedpython:python-restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:restrictedpython:python_restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_dev_project:python-restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_dev_project:python_restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_devproject:python-restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_devproject:python_restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:restrictedpython:restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_dev_project:restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope-dev:python-restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope-dev:python_restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_dev:python-restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_dev:python_restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_devproject:restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python-restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python_restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope-dev:restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:zope_dev:restrictedpython:7.3:*:*:*:*:*:*:*","cpe:2.3:a:python:restrictedpython:7.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/restrictedpython@7.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-27mf-ghqm-j3j8","dataSource":"https://github.com/advisories/GHSA-27mf-ghqm-j3j8","namespace":"github:language:python","severity":"Medium","urls":[],"description":"aiohttp has a memory leak when middleware is enabled when requesting a resource with a non-allowed method","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-52303","epss":0.00088,"percentile":0.26555,"date":"2025-06-02"}],"fix":{"versions":["3.10.11"],"state":"fixed"},"advisories":[],"risk":0.055},"relatedVulnerabilities":[{"id":"CVE-2024-52303","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-52303","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/aio-libs/aiohttp/commit/bc15db61615079d1b6327ba42c682f758fa96936","https://github.com/aio-libs/aiohttp/security/advisories/GHSA-27mf-ghqm-j3j8"],"description":"aiohttp is an asynchronous HTTP client/server framework for asyncio and Python. In versions starting with 3.10.6 and prior to 3.10.11, a memory leak can occur when a request produces a MatchInfoError. This was caused by adding an entry to a cache on each request, due to the building of each MatchInfoError producing a unique cache entry. An attacker may be able to exhaust the memory resources of a server by sending a substantial number (100,000s to millions) of such requests. Those who use any middlewares with aiohttp.web should upgrade to version 3.10.11 to receive a patch.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:N/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":8.7},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-52303","epss":0.00088,"percentile":0.26555,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"aiohttp","version":"3.10.8"}},"found":{"versionConstraint":">=3.10.6,<3.10.11 (python)","vulnerabilityID":"GHSA-27mf-ghqm-j3j8"},"fix":{"suggestedVersion":"3.10.11"}}],"artifact":{"id":"23a9da734e9a4097","name":"aiohttp","version":"3.10.8","type":"python","locations":[{"path":"/lib/python3.12/site-packages/aiohttp-3.10.8.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/aiohttp-3.10.8.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/aiohttp-3.10.8.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/aiohttp-3.10.8.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.12/site-packages/aiohttp-3.10.8.dist-info/top_level.txt","accessPath":"/lib/python3.12/site-packages/aiohttp-3.10.8.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["Apache 2"],"cpes":["cpe:2.3:a:python-aiohttp:python-aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python-aiohttp:python_aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python_aiohttp:python-aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python_aiohttp:python_aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:aiohttp:python-aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:aiohttp:python_aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python-aiohttp:aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python_aiohttp:aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python:python-aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python:python_aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:aiohttp:aiohttp:3.10.8:*:*:*:*:*:*:*","cpe:2.3:a:python:aiohttp:3.10.8:*:*:*:*:*:*:*"],"purl":"pkg:pypi/aiohttp@3.10.8","upstreams":[]}},{"vulnerability":{"id":"GHSA-gmj6-6f8f-6699","dataSource":"https://github.com/advisories/GHSA-gmj6-6f8f-6699","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja has a sandbox breakout through malicious filenames","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56201","epss":0.00064,"percentile":0.20401,"date":"2025-06-02"}],"fix":{"versions":["3.1.5"],"state":"fixed"},"advisories":[],"risk":0.044160000000000005},"relatedVulnerabilities":[{"id":"CVE-2024-56201","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-56201","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/767b23617628419ae3709ccfb02f9602ae9fe51f","https://github.com/pallets/jinja/issues/1792","https://github.com/pallets/jinja/releases/tag/3.1.5","https://github.com/pallets/jinja/security/advisories/GHSA-gmj6-6f8f-6699"],"description":"Jinja is an extensible templating engine. In versions on the 3.x branch prior to 3.1.5, a bug in the Jinja compiler allows an attacker that controls both the content and filename of a template to execute arbitrary Python code, regardless of if Jinja's sandbox is used. To exploit the vulnerability, an attacker needs to control both the filename and the contents of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates where the template author can also choose the template filename. This vulnerability is fixed in 3.1.5.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:P/PR:L/UI:P/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":5.4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56201","epss":0.00064,"percentile":0.20401,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.1.4"}},"found":{"versionConstraint":">=3.0.0,<=3.1.4 (python)","vulnerabilityID":"GHSA-gmj6-6f8f-6699"},"fix":{"suggestedVersion":"3.1.5"}}],"artifact":{"id":"3119b2b2848d353f","name":"jinja2","version":"3.1.4","type":"python","locations":[{"path":"/lib/python3.12/site-packages/jinja2-3.1.4.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/jinja2-3.1.4.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/jinja2-3.1.4.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/jinja2-3.1.4.dist-info/RECORD","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:python-jinja2:python-jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.1.4:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.1.4","upstreams":[]}},{"vulnerability":{"id":"GHSA-cjwg-qfpm-7377","dataSource":"https://github.com/advisories/GHSA-cjwg-qfpm-7377","namespace":"github:language:python","severity":"Medium","urls":[],"description":"python-jose denial of service via compressed JWE content","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":5.3,"exploitabilityScore":3.9,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-33664","epss":0.00081,"percentile":0.2495,"date":"2025-06-02"}],"fix":{"versions":["3.4.0"],"state":"fixed"},"advisories":[],"risk":0.041714999999999995},"relatedVulnerabilities":[{"id":"CVE-2024-33664","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-33664","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/mpdavis/python-jose/issues/344","https://github.com/mpdavis/python-jose/pull/345","https://www.vicarius.io/vsociety/posts/jwt-bomb-in-python-jose-cve-2024-33664","https://github.com/mpdavis/python-jose/issues/344","https://github.com/mpdavis/python-jose/pull/345","https://www.vicarius.io/vsociety/posts/jwt-bomb-in-python-jose-cve-2024-33664"],"description":"python-jose through 3.3.0 allows attackers to cause a denial of service (resource consumption) during a decode via a crafted JSON Web Encryption (JWE) token with a high compression ratio, aka a \"JWT bomb.\" This is similar to CVE-2024-21319.","cvss":[{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":5.3,"exploitabilityScore":3.9,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-33664","epss":0.00081,"percentile":0.2495,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"python-jose","version":"3.3.0"}},"found":{"versionConstraint":"<3.4.0 (python)","vulnerabilityID":"GHSA-cjwg-qfpm-7377"},"fix":{"suggestedVersion":"3.4.0"}}],"artifact":{"id":"b90f868de7b027be","name":"python-jose","version":"3.3.0","type":"python","locations":[{"path":"/lib/python3.12/site-packages/python_jose-3.3.0.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/python_jose-3.3.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/python_jose-3.3.0.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/python_jose-3.3.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.12/site-packages/python_jose-3.3.0.dist-info/top_level.txt","accessPath":"/lib/python3.12/site-packages/python_jose-3.3.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:mike_philip_davis_project:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike_philip_davis_project:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike_philip_davisproject:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike_philip_davisproject:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:michael_davis_project:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:michael_davis_project:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:michael_davisproject:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:michael_davisproject:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike-philip-davis:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike-philip-davis:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike_philip_davis:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike_philip_davis:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:michael_davis:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:michael_davis:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:python-jose:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:python-jose:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:python_jose:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:python_jose:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jose:3.3.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/python-jose@3.3.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-vqfr-h8mv-ghfj","dataSource":"https://github.com/advisories/GHSA-vqfr-h8mv-ghfj","namespace":"github:language:python","severity":"Critical","urls":[],"description":"h11 accepts some malformed Chunked-Encoding bodies","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:N","metrics":{"baseScore":9.1,"exploitabilityScore":3.9,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-43859","epss":0.00044,"percentile":0.13131,"date":"2025-06-02"}],"fix":{"versions":["0.16.0"],"state":"fixed"},"advisories":[],"risk":0.03982},"relatedVulnerabilities":[{"id":"CVE-2025-43859","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-43859","namespace":"nvd:cpe","severity":"Critical","urls":["https://github.com/python-hyper/h11/commit/114803a29ce50116dc47951c690ad4892b1a36ed","https://github.com/python-hyper/h11/security/advisories/GHSA-vqfr-h8mv-ghfj"],"description":"h11 is a Python implementation of HTTP/1.1. Prior to version 0.16.0, a leniency in h11's parsing of line terminators in chunked-coding message bodies can lead to request smuggling vulnerabilities under certain conditions. This issue has been patched in version 0.16.0. Since exploitation requires the combination of buggy h11 with a buggy (reverse) proxy, fixing either component is sufficient to mitigate this issue.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:N","metrics":{"baseScore":9.1,"exploitabilityScore":3.9,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-43859","epss":0.00044,"percentile":0.13131,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"h11","version":"0.14.0"}},"found":{"versionConstraint":"<0.16.0 (python)","vulnerabilityID":"GHSA-vqfr-h8mv-ghfj"},"fix":{"suggestedVersion":"0.16.0"}}],"artifact":{"id":"a1550f9cb269817c","name":"h11","version":"0.14.0","type":"python","locations":[{"path":"/lib/python3.12/site-packages/h11-0.14.0.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/h11-0.14.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/h11-0.14.0.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/h11-0.14.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.12/site-packages/h11-0.14.0.dist-info/top_level.txt","accessPath":"/lib/python3.12/site-packages/h11-0.14.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:nathaniel_j__smith_project:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smith_project:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smithproject:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smithproject:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smith_project:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smith:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smith:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smithproject:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smith:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njs_project:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njs_project:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njsproject:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njsproject:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python-h11:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python-h11:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python_h11:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python_h11:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njs_project:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:h11:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:h11:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njs:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njs:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njsproject:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python-h11:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python_h11:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:h11:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njs:h11:0.14.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/h11@0.14.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-79v4-65xg-pq4g","dataSource":"https://github.com/advisories/GHSA-79v4-65xg-pq4g","namespace":"github:language:python","severity":"Low","urls":[],"description":"Vulnerable OpenSSL included in cryptography wheels","cvss":[],"epss":[{"cve":"CVE-2024-12797","epss":0.0013,"percentile":0.33813,"date":"2025-06-02"}],"fix":{"versions":["44.0.1"],"state":"fixed"},"advisories":[],"risk":0.039},"relatedVulnerabilities":[{"id":"CVE-2024-12797","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-12797","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/openssl/openssl/commit/738d4f9fdeaad57660dcba50a619fafced3fd5e9","https://github.com/openssl/openssl/commit/798779d43494549b611233f92652f0da5328fbe7","https://github.com/openssl/openssl/commit/87ebd203feffcf92ad5889df92f90bb0ee10a699","https://openssl-library.org/news/secadv/20250211.txt","http://www.openwall.com/lists/oss-security/2025/02/11/3","http://www.openwall.com/lists/oss-security/2025/02/11/4","https://security.netapp.com/advisory/ntap-20250214-0001/"],"description":"Issue summary: Clients using RFC7250 Raw Public Keys (RPKs) to authenticate a\nserver may fail to notice that the server was not authenticated, because\nhandshakes don't abort as expected when the SSL_VERIFY_PEER verification mode\nis set.\n\nImpact summary: TLS and DTLS connections using raw public keys may be\nvulnerable to man-in-middle attacks when server authentication failure is not\ndetected by clients.\n\nRPKs are disabled by default in both TLS clients and TLS servers.  The issue\nonly arises when TLS clients explicitly enable RPK use by the server, and the\nserver, likewise, enables sending of an RPK instead of an X.509 certificate\nchain.  The affected clients are those that then rely on the handshake to\nfail when the server's RPK fails to match one of the expected public keys,\nby setting the verification mode to SSL_VERIFY_PEER.\n\nClients that enable server-side raw public keys can still find out that raw\npublic key verification failed by calling SSL_get_verify_result(), and those\nthat do, and take appropriate action, are not affected.  This issue was\nintroduced in the initial implementation of RPK support in OpenSSL 3.2.\n\nThe FIPS modules in 3.4, 3.3, 3.2, 3.1 and 3.0 are not affected by this issue.","cvss":[{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:L","metrics":{"baseScore":6.3,"exploitabilityScore":2.9,"impactScore":3.4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-12797","epss":0.0013,"percentile":0.33813,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"cryptography","version":"43.0.1"}},"found":{"versionConstraint":">=42.0.0,<44.0.1 (python)","vulnerabilityID":"GHSA-79v4-65xg-pq4g"},"fix":{"suggestedVersion":"44.0.1"}}],"artifact":{"id":"6bb7964ba3c434f6","name":"cryptography","version":"43.0.1","type":"python","locations":[{"path":"/lib/python3.12/site-packages/cryptography-43.0.1.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/cryptography-43.0.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/cryptography-43.0.1.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/cryptography-43.0.1.dist-info/RECORD","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["Apache-2.0 OR BSD-3-Clause"],"cpes":["cpe:2.3:a:cryptography.io:cryptography:43.0.1:*:*:*:*:python:*:*","cpe:2.3:a:cryptography.io:cryptography:43.0.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/cryptography@43.0.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-pwhh-q4h6-w599","dataSource":"https://github.com/advisories/GHSA-pwhh-q4h6-w599","namespace":"github:language:python","severity":"High","urls":[],"description":"Spotipy's cache file, containing spotify auth token, is created with overly broad permissions","cvss":[],"epss":[{"cve":"CVE-2025-27154","epss":0.00048,"percentile":0.14641,"date":"2025-06-02"}],"fix":{"versions":["2.25.1"],"state":"fixed"},"advisories":[],"risk":0.036000000000000004},"relatedVulnerabilities":[{"id":"CVE-2025-27154","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-27154","namespace":"nvd:cpe","severity":"Critical","urls":["https://github.com/spotipy-dev/spotipy/blob/master/spotipy/cache_handler.py#L93-L98","https://github.com/spotipy-dev/spotipy/commit/1ca453f6ef87a2a9e9876f52b6cb38d13532ccf2","https://github.com/spotipy-dev/spotipy/releases/tag/2.25.1","https://github.com/spotipy-dev/spotipy/security/advisories/GHSA-pwhh-q4h6-w599","https://github.com/spotipy-dev/spotipy/security/advisories/GHSA-pwhh-q4h6-w599"],"description":"Spotipy is a lightweight Python library for the Spotify Web API. The `CacheHandler` class creates a cache file to store the auth token. Prior to version 2.25.1, the file created has `rw-r--r--` (644) permissions by default, when it could be locked down to `rw-------` (600) permissions. This leads to overly broad exposure of the spotify auth token. If this token can be read by an attacker (another user on the machine, or a process running as another user), it can be used to perform administrative actions on the Spotify account, depending on the scope granted to the token. Version 2.25.1 tightens the cache file permissions.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":9.8,"exploitabilityScore":3.9,"impactScore":5.9},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:N/PR:L/UI:N/VC:H/VI:H/VA:N/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":8.4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-27154","epss":0.00048,"percentile":0.14641,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"spotipy","version":"2.23.0"}},"found":{"versionConstraint":"<2.25.1 (python)","vulnerabilityID":"GHSA-pwhh-q4h6-w599"},"fix":{"suggestedVersion":"2.25.1"}}],"artifact":{"id":"56df8518428a5696","name":"spotipy","version":"2.23.0","type":"python","locations":[{"path":"/lib/python3.12/site-packages/spotipy-2.23.0.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/spotipy-2.23.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/spotipy-2.23.0.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/spotipy-2.23.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.12/site-packages/spotipy-2.23.0.dist-info/top_level.txt","accessPath":"/lib/python3.12/site-packages/spotipy-2.23.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:@plamere_project:python-spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:@plamere_project:python_spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:@plamereproject:python-spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:@plamereproject:python_spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:python-spotipy:python-spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:python-spotipy:python_spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:python_spotipy:python-spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:python_spotipy:python_spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:paul_project:python-spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:paul_project:python_spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:paulproject:python-spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:paulproject:python_spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:@plamere_project:spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:@plamere:python-spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:@plamere:python_spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:@plamereproject:spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:python-spotipy:spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:python_spotipy:spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:spotipy:python-spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:spotipy:python_spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:paul_project:spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:paul:python-spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:paul:python_spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:paulproject:spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:@plamere:spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:spotipy:spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:python:spotipy:2.23.0:*:*:*:*:*:*:*","cpe:2.3:a:paul:spotipy:2.23.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/spotipy@2.23.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-6c5p-j8vq-pqhj","dataSource":"https://github.com/advisories/GHSA-6c5p-j8vq-pqhj","namespace":"github:language:python","severity":"Critical","urls":[],"description":"python-jose algorithm confusion with OpenSSH ECDSA keys","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:N","metrics":{"baseScore":7.4,"exploitabilityScore":2.3,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-33663","epss":0.00041,"percentile":0.11895,"date":"2025-06-02"}],"fix":{"versions":["3.4.0"],"state":"fixed"},"advisories":[],"risk":0.033620000000000004},"relatedVulnerabilities":[{"id":"CVE-2024-33663","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-33663","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/mpdavis/python-jose/issues/346","https://www.vicarius.io/vsociety/posts/algorithm-confusion-in-python-jose-cve-2024-33663","https://github.com/mpdavis/python-jose/issues/346","https://www.vicarius.io/vsociety/posts/algorithm-confusion-in-python-jose-cve-2024-33663"],"description":"python-jose through 3.3.0 has algorithm confusion with OpenSSH ECDSA keys and other key formats. This is similar to CVE-2022-29217.","cvss":[{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:L/A:N","metrics":{"baseScore":6.5,"exploitabilityScore":3.9,"impactScore":2.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-33663","epss":0.00041,"percentile":0.11895,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"python-jose","version":"3.3.0"}},"found":{"versionConstraint":"<3.4.0 (python)","vulnerabilityID":"GHSA-6c5p-j8vq-pqhj"},"fix":{"suggestedVersion":"3.4.0"}}],"artifact":{"id":"b90f868de7b027be","name":"python-jose","version":"3.3.0","type":"python","locations":[{"path":"/lib/python3.12/site-packages/python_jose-3.3.0.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/python_jose-3.3.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/python_jose-3.3.0.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/python_jose-3.3.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.12/site-packages/python_jose-3.3.0.dist-info/top_level.txt","accessPath":"/lib/python3.12/site-packages/python_jose-3.3.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:mike_philip_davis_project:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike_philip_davis_project:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike_philip_davisproject:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike_philip_davisproject:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:michael_davis_project:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:michael_davis_project:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:michael_davisproject:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:michael_davisproject:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike-philip-davis:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike-philip-davis:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike_philip_davis:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:mike_philip_davis:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:michael_davis:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:michael_davis:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:python-jose:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:python-jose:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:python_jose:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:python_jose:python_jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jose:3.3.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jose:3.3.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/python-jose@3.3.0","upstreams":[]}},{"vulnerability":{"id":"CVE-2024-9287","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-9287","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/python/cpython/commit/633555735a023d3e4d92ba31da35b1205f9ecbd7","https://github.com/python/cpython/commit/8450b2482586857d689b6658f08de9c8179af7db","https://github.com/python/cpython/commit/9286ab3a107ea41bd3f3c3682ce2512692bdded8","https://github.com/python/cpython/commit/ae961ae94bf19c8f8c7fbea3d1c25cc55ce8ae97","https://github.com/python/cpython/commit/d48cc82ed25e26b02eb97c6263d95dcaa1e9111b","https://github.com/python/cpython/commit/e52095a0c1005a87eed2276af7a1f2f66e2b6483","https://github.com/python/cpython/issues/124651","https://github.com/python/cpython/pull/124712","https://mail.python.org/archives/list/security-announce@python.org/thread/RSPJ2B5JL22FG3TKUJ7D7DQ4N5JRRBZL/","https://security.netapp.com/advisory/ntap-20250425-0006/"],"description":"A vulnerability has been found in the CPython `venv` module and CLI where path names provided when creating a virtual environment were not quoted properly, allowing the creator to inject commands into virtual environment \"activation\" scripts (ie \"source venv/bin/activate\"). This means that attacker-controlled virtual environments are able to run commands when the virtual environment is activated. Virtual environments which are not created by an attacker or which aren't activated before being used (ie \"./venv/bin/python\") are not affected.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}},{"source":"cna@python.org","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:P/PR:H/UI:A/VC:H/VI:H/VA:N/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:Green","metrics":{"baseScore":5.3},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-9287","epss":0.00038,"percentile":0.10564,"date":"2025-06-02"}],"fix":{"versions":["3.9.21","3.10.16","3.11.11","3.12.8","3.13.1","3.14.0a2"],"state":"fixed"},"advisories":[],"risk":0.026695},"relatedVulnerabilities":[],"matchDetails":[{"type":"cpe-match","matcher":"stock-matcher","searchedBy":{"namespace":"nvd:cpe","cpes":["cpe:2.3:a:python:python:3.13.0:*:*:*:*:*:*:*"],"package":{"name":"python","version":"3.13.0"}},"found":{"vulnerabilityID":"CVE-2024-9287","versionConstraint":"< 3.9.21,>= 3.10.0, < 3.10.16,>= 3.11.0, < 3.11.11,>= 3.12.0, < 3.12.8,>= 3.13.0, < 3.13.1,>= 3.14.0a1, < 3.14.0a2 (unknown)","cpes":["cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"]},"fix":{"suggestedVersion":"3.13.1"}}],"artifact":{"id":"47bfec9b02a55f84","name":"python","version":"3.13.0","type":"binary","locations":[{"path":"/usr/bin/python3.13","accessPath":"/usr/bin/python3.13","annotations":{"evidence":"primary"}}],"language":"","licenses":[],"cpes":["cpe:2.3:a:python_software_foundation:python:3.13.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python:3.13.0:*:*:*:*:*:*:*"],"purl":"pkg:generic/python@3.13.0","upstreams":[]}},{"vulnerability":{"id":"CVE-2024-3220","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-3220","namespace":"nvd:cpe","severity":"Low","urls":["https://mail.python.org/archives/list/security-announce@python.org/thread/CDXW34ND2LSAOYAR5N6UNONP4ZBX4D6R/","http://www.openwall.com/lists/oss-security/2025/02/14/8","https://security.netapp.com/advisory/ntap-20250314-0001/"],"description":"There is a defect in the CPython standard library module “mimetypes” where on Windows the default list of known file locations are writable meaning other users can create invalid files to cause MemoryError to be raised on Python runtime startup or have file extensions be interpreted as the incorrect file type.\n\nThis defect is caused by the default locations of Linux and macOS platforms (such as “/etc/mime.types”) also being used on Windows, where they are user-writable locations (“C:\\etc\\mime.types”).\n\nTo work-around this issue a user can call mimetypes.init() with an empty list (“[]”) on Windows platforms to avoid using the default list of known file locations.","cvss":[{"source":"cna@python.org","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:P/PR:L/UI:N/VC:N/VI:L/VA:L/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":2.3},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-3220","epss":0.00088,"percentile":0.26555,"date":"2025-06-02"}],"fix":{"versions":["3.14.0"],"state":"fixed"},"advisories":[],"risk":0.023320000000000004},"relatedVulnerabilities":[],"matchDetails":[{"type":"cpe-match","matcher":"stock-matcher","searchedBy":{"namespace":"nvd:cpe","cpes":["cpe:2.3:a:python:python:3.13.0:*:*:*:*:*:*:*"],"package":{"name":"python","version":"3.13.0"}},"found":{"vulnerabilityID":"CVE-2024-3220","versionConstraint":"< 3.14.0 (unknown)","cpes":["cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"]},"fix":{"suggestedVersion":"3.14.0"}}],"artifact":{"id":"47bfec9b02a55f84","name":"python","version":"3.13.0","type":"binary","locations":[{"path":"/usr/bin/python3.13","accessPath":"/usr/bin/python3.13","annotations":{"evidence":"primary"}}],"language":"","licenses":[],"cpes":["cpe:2.3:a:python_software_foundation:python:3.13.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python:3.13.0:*:*:*:*:*:*:*"],"purl":"pkg:generic/python@3.13.0","upstreams":[]}},{"vulnerability":{"id":"CVE-2024-50602","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-50602","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/libexpat/libexpat/pull/915","https://lists.debian.org/debian-lts-announce/2025/04/msg00040.html","https://security.netapp.com/advisory/ntap-20250404-0008/"],"description":"An issue was discovered in libexpat before 2.6.4. There is a crash within the XML_ResumeParser function because XML_StopParser can stop/suspend an unstarted parser.","cvss":[{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.9,"exploitabilityScore":2.3,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-50602","epss":0.00037,"percentile":0.10035,"date":"2025-06-02"}],"fix":{"versions":["3.9.21","3.10.16","3.11.11","3.12.8","3.13.1"],"state":"fixed"},"advisories":[],"risk":0.020165000000000002},"relatedVulnerabilities":[],"matchDetails":[{"type":"cpe-match","matcher":"stock-matcher","searchedBy":{"namespace":"nvd:cpe","cpes":["cpe:2.3:a:python:python:3.13.0:*:*:*:*:*:*:*"],"package":{"name":"python","version":"3.13.0"}},"found":{"vulnerabilityID":"CVE-2024-50602","versionConstraint":"< 3.9.21,>= 3.10, < 3.10.16,>= 3.11, < 3.11.11,>= 3.12, < 3.12.8,>= 3.13, < 3.13.1 (unknown)","cpes":["cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"]},"fix":{"suggestedVersion":"3.13.1"}}],"artifact":{"id":"47bfec9b02a55f84","name":"python","version":"3.13.0","type":"binary","locations":[{"path":"/usr/bin/python3.13","accessPath":"/usr/bin/python3.13","annotations":{"evidence":"primary"}}],"language":"","licenses":[],"cpes":["cpe:2.3:a:python_software_foundation:python:3.13.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python:3.13.0:*:*:*:*:*:*:*"],"purl":"pkg:generic/python@3.13.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-cpwx-vrp4-4pq7","dataSource":"https://github.com/advisories/GHSA-cpwx-vrp4-4pq7","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja2 vulnerable to sandbox breakout through attr filter selecting format method","cvss":[],"epss":[{"cve":"CVE-2025-27516","epss":0.00034,"percentile":0.08503,"date":"2025-06-02"}],"fix":{"versions":["3.1.6"],"state":"fixed"},"advisories":[],"risk":0.017},"relatedVulnerabilities":[{"id":"CVE-2025-27516","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-27516","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/90457bbf33b8662926ae65cdde4c4c32e756e403","https://github.com/pallets/jinja/security/advisories/GHSA-cpwx-vrp4-4pq7","https://lists.debian.org/debian-lts-announce/2025/04/msg00045.html"],"description":"Jinja is an extensible templating engine. Prior to 3.1.6, an oversight in how the Jinja sandboxed environment interacts with the |attr filter allows an attacker that controls the content of a template to execute arbitrary Python code. To exploit the vulnerability, an attacker needs to control the content of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates. Jinja's sandbox does catch calls to str.format and ensures they don't escape the sandbox. However, it's possible to use the |attr filter to get a reference to a string's plain format method, bypassing the sandbox. After the fix, the |attr filter no longer bypasses the environment's attribute lookup. This vulnerability is fixed in 3.1.6.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:P/PR:L/UI:P/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":5.4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-27516","epss":0.00034,"percentile":0.08503,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.1.4"}},"found":{"versionConstraint":"<=3.1.5 (python)","vulnerabilityID":"GHSA-cpwx-vrp4-4pq7"},"fix":{"suggestedVersion":"3.1.6"}}],"artifact":{"id":"3119b2b2848d353f","name":"jinja2","version":"3.1.4","type":"python","locations":[{"path":"/lib/python3.12/site-packages/jinja2-3.1.4.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/jinja2-3.1.4.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/jinja2-3.1.4.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/jinja2-3.1.4.dist-info/RECORD","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:python-jinja2:python-jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.1.4:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.1.4:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.1.4","upstreams":[]}},{"vulnerability":{"id":"GHSA-g92j-qhmh-64v2","dataSource":"https://github.com/advisories/GHSA-g92j-qhmh-64v2","namespace":"github:language:python","severity":"Low","urls":[],"description":"Sentry's Python SDK unintentionally exposes environment variables to subprocesses","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:H/UI:N/S:C/C:L/I:N/A:N","metrics":{"baseScore":2.5,"exploitabilityScore":0.9,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-40647","epss":0.00045,"percentile":0.13392,"date":"2025-06-02"}],"fix":{"versions":["2.8.0"],"state":"fixed"},"advisories":[],"risk":0.012375},"relatedVulnerabilities":[{"id":"CVE-2024-40647","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-40647","namespace":"nvd:cpe","severity":"Medium","urls":["https://docs.python.org/3/library/subprocess.html","https://docs.sentry.io/platforms/python/integrations/default-integrations","https://docs.sentry.io/platforms/python/integrations/default-integrations/#stdlib","https://github.com/getsentry/sentry-python/commit/763e40aa4cb57ecced467f48f78f335c87e9bdff","https://github.com/getsentry/sentry-python/pull/3251","https://github.com/getsentry/sentry-python/releases/tag/2.8.0","https://github.com/getsentry/sentry-python/security/advisories/GHSA-g92j-qhmh-64v2","https://docs.python.org/3/library/subprocess.html","https://docs.sentry.io/platforms/python/integrations/default-integrations","https://docs.sentry.io/platforms/python/integrations/default-integrations/#stdlib","https://github.com/getsentry/sentry-python/commit/763e40aa4cb57ecced467f48f78f335c87e9bdff","https://github.com/getsentry/sentry-python/pull/3251","https://github.com/getsentry/sentry-python/releases/tag/2.8.0","https://github.com/getsentry/sentry-python/security/advisories/GHSA-g92j-qhmh-64v2"],"description":"sentry-sdk is the official Python SDK for Sentry.io. A bug in Sentry's Python SDK < 2.8.0 allows the environment variables to be passed to subprocesses despite the `env={}` setting. In Python's `subprocess` calls, all environment variables are passed to subprocesses by default. However, if you specifically do not want them to be passed to subprocesses, you may use `env` argument in `subprocess` calls. Due to the bug in Sentry SDK, with the Stdlib integration enabled (which is enabled by default), this expectation is not fulfilled, and all environment variables are being passed to subprocesses instead. The issue has been patched in pull request #3251 and is included in sentry-sdk==2.8.0. We strongly recommend upgrading to the latest SDK version. However, if it's not possible, and if passing environment variables to child processes poses a security risk for you, you can disable all default integrations.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:H/UI:N/S:C/C:H/I:N/A:N","metrics":{"baseScore":5.3,"exploitabilityScore":0.9,"impactScore":4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-40647","epss":0.00045,"percentile":0.13392,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"sentry-sdk","version":"1.40.3"}},"found":{"versionConstraint":"<2.8.0 (python)","vulnerabilityID":"GHSA-g92j-qhmh-64v2"},"fix":{"suggestedVersion":"2.8.0"}}],"artifact":{"id":"566ac63e637ad9ba","name":"sentry-sdk","version":"1.40.3","type":"python","locations":[{"path":"/lib/python3.12/site-packages/sentry_sdk-1.40.3.dist-info/METADATA","accessPath":"/lib/python3.12/site-packages/sentry_sdk-1.40.3.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.12/site-packages/sentry_sdk-1.40.3.dist-info/RECORD","accessPath":"/lib/python3.12/site-packages/sentry_sdk-1.40.3.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.12/site-packages/sentry_sdk-1.40.3.dist-info/top_level.txt","accessPath":"/lib/python3.12/site-packages/sentry_sdk-1.40.3.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:sentry:sentry_software_development_kit:1.40.3:*:*:*:*:python:*:*"],"purl":"pkg:pypi/sentry-sdk@1.40.3","upstreams":[]}},{"vulnerability":{"id":"CVE-2025-4516","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-4516","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/python/cpython/commit/69b4387f78f413e8c47572a85b3478c47eba8142","https://github.com/python/cpython/commit/9f69a58623bd01349a18ba0c7a9cb1dad6a51e8e","https://github.com/python/cpython/issues/133767","https://github.com/python/cpython/pull/129648","https://mail.python.org/archives/list/security-announce@python.org/thread/L75IPBBTSCYEF56I2M4KIW353BB3AY74/","http://www.openwall.com/lists/oss-security/2025/05/16/4","http://www.openwall.com/lists/oss-security/2025/05/19/1"],"description":"There is an issue in CPython when using `bytes.decode(\"unicode_escape\", error=\"ignore|replace\")`. If you are not using the \"unicode_escape\" encoding or an error handler your usage is not affected. To work-around this issue you may stop using the error= handler and instead wrap the bytes.decode() call in a try-except catching the DecodeError.","cvss":[{"source":"cna@python.org","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:H/AT:P/PR:N/UI:N/VC:N/VI:N/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-4516","epss":0.00018,"percentile":0.03232,"date":"2025-06-02"}],"fix":{"versions":["3.14.0b1"],"state":"fixed"},"advisories":[],"risk":0.009810000000000001},"relatedVulnerabilities":[],"matchDetails":[{"type":"cpe-match","matcher":"stock-matcher","searchedBy":{"namespace":"nvd:cpe","cpes":["cpe:2.3:a:python:python:3.13.0:*:*:*:*:*:*:*"],"package":{"name":"python","version":"3.13.0"}},"found":{"vulnerabilityID":"CVE-2025-4516","versionConstraint":"< 3.14.0b1 (unknown)","cpes":["cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"]},"fix":{"suggestedVersion":"3.14.0b1"}}],"artifact":{"id":"47bfec9b02a55f84","name":"python","version":"3.13.0","type":"binary","locations":[{"path":"/usr/bin/python3.13","accessPath":"/usr/bin/python3.13","annotations":{"evidence":"primary"}}],"language":"","licenses":[],"cpes":["cpe:2.3:a:python_software_foundation:python:3.13.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python:3.13.0:*:*:*:*:*:*:*"],"purl":"pkg:generic/python@3.13.0","upstreams":[]}}],"source":{"type":"sbom-file","target":"home-assistant-snap-syft.json"},"distro":{"name":"","version":"","idLike":null},"descriptor":{"name":"grype","version":"0.92.2","configuration":{"output":["json=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-05-21/home-assistant-snap-grype.json","template=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-05-21/home-assistant-snap-grype.html"],"file":"","pretty":false,"distro":"","add-cpes-if-none":false,"output-template-file":"../html.tmpl","check-for-app-update":true,"only-fixed":false,"only-notfixed":false,"ignore-wontfix":"","platform":"","search":{"scope":"squashed","unindexed-archives":false,"indexed-archives":true},"ignore":[{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"kernel-headers","version":"","language":"","type":"rpm","location":"","upstream-name":"kernel"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux(-.*)?-headers-.*","version":"","language":"","type":"deb","location":"","upstream-name":"linux.*"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux-libc-dev","version":"","language":"","type":"deb","location":"","upstream-name":"linux"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"}],"exclude":[],"externalSources":{"enable":false,"maven":{"searchUpstreamBySha1":true,"baseUrl":"https://search.maven.org/solrsearch/select","rateLimit":300000000}},"match":{"java":{"using-cpes":false},"jvm":{"using-cpes":true},"dotnet":{"using-cpes":false},"golang":{"using-cpes":false,"always-use-cpe-for-stdlib":true,"allow-main-module-pseudo-version-comparison":false},"javascript":{"using-cpes":false},"python":{"using-cpes":false},"ruby":{"using-cpes":false},"rust":{"using-cpes":false},"stock":{"using-cpes":true}},"fail-on-severity":"","registry":{"insecure-skip-tls-verify":false,"insecure-use-http":false,"auth":null,"ca-cert":""},"show-suppressed":false,"by-cve":false,"SortBy":{"sort-by":"risk"},"name":"","default-image-pull-source":"","vex-documents":[],"vex-add":[],"match-upstream-kernel-headers":false,"db":{"cache-dir":"/Users/alan/Library/Caches/grype/db","update-url":"https://grype.anchore.io/databases","ca-cert":"","auto-update":true,"validate-by-hash-on-start":true,"validate-age":true,"max-allowed-built-age":432000000000000,"require-update-check":false,"update-available-timeout":30000000000,"update-download-timeout":300000000000,"max-update-check-frequency":7200000000000},"exp":{},"dev":{"db":{"debug":false}}},"db":{"status":{"schemaVersion":"v6.0.2","from":"https://grype.anchore.io/databases/v6/vulnerability-db_v6.0.2_2025-06-03T01:32:03Z_1748923962.tar.zst?checksum=sha256%3A02a6127217518a93e0ef8739cb9f2d659109fff0d9cc25f0c13d5ca8b307a118","built":"2025-06-03T04:12:42Z","path":"/Users/alan/Library/Caches/grype/db/6/vulnerability.db","valid":true},"providers":{"alpine":{"captured":"2025-06-03T01:32:27Z","input":"xxh64:a6a065ece392f6d1"},"amazon":{"captured":"2025-06-03T01:32:10Z","input":"xxh64:b65d458d6858d72a"},"chainguard":{"captured":"2025-06-03T01:32:15Z","input":"xxh64:bad092583f78d4dd"},"debian":{"captured":"2025-06-03T01:32:22Z","input":"xxh64:b4717cad509f50d8"},"epss":{"captured":"2025-06-03T01:32:43Z","input":"xxh64:2f9660be834e0ce6"},"github":{"captured":"2025-06-03T01:32:21Z","input":"xxh64:cdd94f1952fcf607"},"kev":{"captured":"2025-06-03T01:32:03Z","input":"xxh64:575844c3434cc70b"},"mariner":{"captured":"2025-06-03T01:32:11Z","input":"xxh64:7628f1adb0b9b12b"},"nvd":{"captured":"2025-06-03T01:32:30Z","input":"xxh64:463c4e9a1d401101"},"oracle":{"captured":"2025-06-03T01:32:27Z","input":"xxh64:257df3059d28d823"},"rhel":{"captured":"2025-06-03T01:33:04Z","input":"xxh64:59e7d70bf674afdc"},"sles":{"captured":"2025-06-03T01:32:46Z","input":"xxh64:3cf7f372ea0c981a"},"ubuntu":{"captured":"2025-06-03T01:33:20Z","input":"xxh64:075be8afc080f2be"},"wolfi":{"captured":"2025-06-03T01:32:31Z","input":"xxh64:1aa0b26b004bdd38"}}},"timestamp":"2025-06-03T13:05:55.341652+01:00"}}
